python-client/giskard/models/__init__.py

```diff
@@ -0,0 +1,153 @@
+from importlib import import_module
+import torch  # TODO: to be omitted in another PR
+from typing import Union
+from giskard.core.core import SupportedModelTypes
+from giskard.models.sklearn import SKLearnModel
+from giskard.models.catboost import CatboostModel
+from giskard.models.pytorch import PyTorchModel
+from giskard.models.tensorflow import TensorFlowModel
+from giskard.models.huggingface import HuggingFaceModel
+
+# format: dict[GiskardModel: list(tuple(module, base_class))]
+_libraries = {HuggingFaceModel: [("transformers", "PreTrainedModel")],
+              SKLearnModel: [("sklearn.base", "BaseEstimator")],
+              CatboostModel: [("catboost", "CatBoost"),
```

Comment:
why not just checking for the parent class `CatBoost` directly? In either case all of the catboost models inherit from it and pass by its `save_model` method when being saved

Reply:
Indeed it's weird that mlflow check each subtype separately, looking at the code, they all inherit from `CatBoost`

https://github.com/catboost/catboost/blob/d84d92c6757c71132b5d700c4e09cafe81f13422/catboost/python-package/catboost/core.py#L1754

Reply:
Oh, sorry. You're right! I didn't check their code actually, only their doc, then mlflow which made me a bit biased. Thanks for this catch!